custom-instrumentation/instrument/index.js

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+const newrelic = require('newrelic')
+const Queue = require('./job-queue')
+
+function exampleJob() {
+    // Do whatever work you want here - this is just a placeholder
+    return 'job done';
+}
+
+function cbJob(cb) {
+    const result = exampleJob();
+    return cb('Callback ' + result);
+}
+
+async function promiseJob() {
+    return new Promise((resolve, reject) => {
+        try {
+            const result = exampleJob()
+            resolve('Promise ' + result)
+        }
+        catch (error) {
+            reject(error)
+        }
+    })
+}
+
+function main(){
+    const queue = new Queue()
+
+    // We will be creating our transacations with startBackgroundTransaction
+    // because we are not operating inside a typical web framework. If you already
+    // are operating within a web framework with transactions, you may omit
+    // the startBackgroundTransaction wrapper.
+    newrelic.startBackgroundTransaction('firstTransaction', function first() {
```

Comment:
```suggestion
    // We will be creating our transacations with startBackgroundTransaction
    // because this application does not utilize frameworks New Relic already
    // instruments. Thus, transactions are not automatically created for use.
    // If you are already operating within an instrumented framework, you may
    // omit the startBackgroundTransaction wrapper.
    newrelic.startBackgroundTransaction('firstTransaction', function first() {
```